### PR TITLE
Add support to configure zookeeper clientPortListenBacklog and server CnxnFactory parameters

### PR DESCRIPTION
**Pull request by amuraru**
### Change log description

Add support to configure zookeeper clientPortListenBacklog and server CnxnFactory parameters

Fixes https://github.com/pravega/zookeeper-operator/issues/356 

### Purpose of the change

In large clusters, increasing clientPortListenBacklog (from default 50) is required to handle large number of clients

### What the code does

Adds two new zk configuration parameters.

### CODE REVIEW

**Comment by anishakj** [comment]
@amuraru Please open a github issue corresponding to this

**Comment by amuraru** [comment]
filed #356

File: pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go [hunk 1]
```diff
@@ -613,6 +613,18 @@ type ZookeeperConfig struct {
 	// The default value is 60
 	MaxClientCnxns int `json:"maxClientCnxns,omitempty"`
 
+	// ClientPortListenBacklog The socket backlog length for the ZooKeeper server socket.
+	// This controls the number of requests that will be queued server-side to be processed by the ZooKeeper server.
+	// Connections that exceed this length will receive a network timeout (30s) which may cause ZooKeeper session expiry
```
Comment (anishakj) [inline pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go:618]: Can we create a map for additional options and parse these options while creating configmap string in `makeZkConfigString`

**Comment by amuraru** [comment]
Closing this in favour of https://github.com/pravega/zookeeper-operator/pull/360

**Pull request closed without merging**